src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -44,7 +44,7 @@ public sealed partial class Pipe
         private readonly PipeScheduler _readerScheduler;
         private readonly PipeScheduler _writerScheduler;
 
-        private readonly Stack<BufferSegment> _bufferSegmentPool;
+        private readonly BufferSegmentStack _bufferSegmentPool;
```

Comment:
How does this work? I thought readonly struct fields are copied before method invocations.

Reply:
Ah frack, I thought I changed this. I'll add a comment too.

Reply:
I'd rather you added a test.